util/stop/stop.go

```diff
@@ -0,0 +1,34 @@
+// Package stop provides mechanism to channel graceful shutdown signal through
+// a hierarchy of running components.
+package stop
+
+// Stopper allows gracefully stoping multiple operations.
+type Stopper chan struct{}
```

Comment:
A fully idea is to name Stopper and Stoppable as `Er` and `Able`, so that they are referenced as `stop.Er` and `stop.Able` from other code.